### PR TITLE
[#3001] docs(spark-connector): add spark connector document

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
add spark connector document

### Why are the changes needed?
Fix: #3001

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
document

### CODE REVIEW

File: docs/spark-connector/spark-connector.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+--conf spark.plugins="com.datastrato.gravitino.spark.connector.plugin.GravitinoSparkPlugin" \
+--conf spark.sql.gravitino.uri=http://127.0.0.1:8090 \
+--conf spark.sql.gravitino.metalake=test \
+--conf spark.sql.warehouse.dir=hdfs://127.0.0.1:9000/user/hive/warehouse-hive
+```
+
+3. Execute the spark sql query. 
+
+Suppose there are two catalogs in the metalake `test`, `hive` and `iceberg`, and the table `hive_table1` in the catalog `hive`, and the table `iceberg_table1` in the catalog `iceberg`.
+```sql
```
Comment (jerryshao) [inline docs/spark-connector/spark-connector.md:47]: add blank line between paragraphs.

File: docs/spark-connector/spark-connector.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+select * from hive.db.hive_table1 union all select * from iceberg.db.iceberg_table1;
+use hive;
+select * from db.hive_table1;
+```
+:::info
```
Comment (jerryshao) [inline docs/spark-connector/spark-connector.md:52]: Also here.

File: docs/spark-connector/spark-connector.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+The command SHOW CATALOGS will only display the Spark default catalog, named spark_catalog, due to limitations within the Spark catalog manager. It does not list the catalogs present in the metalake. However, after explicitly using the USING CATALOG command with a specific catalog name, that catalog name then becomes visible in the output of SHOW CATALOGS.
```
Comment (jerryshao) [inline docs/spark-connector/spark-connector.md:53]: Please use backquotes "`" in "SHOW CATALOGS".

File: docs/spark-connector/spark-catalog-hive.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
+---
+title: "Spark connector hive catalog"
+slug: /spark-connector/spark-catalog-hive
+keyword: spark connector hive catalog
+license: "Copyright 2024 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2."
+---
+
+With the Gravitino Spark connector, accessing data or managing metadata in Hive catalogs becomes straightforward, enabling seamless federation queries across different Hive catalogs.
```
Comment (jerryshao) [inline docs/spark-connector/spark-catalog-hive.md:9]: Do you have some Hive or Iceberg specific configurations that should be listed here?

File: docs/spark-connector/spark-catalog-hive.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    [ OPTIONS ( key1=val1, key2=val2, ... ) ]
+    [ PARTITIONED BY ( col_name1, col_name2, ... ) ]
+    [ CLUSTERED BY ( col_name3, col_name4, ... ) 
+        [ SORTED BY ( col_name [ ASC | DESC ], ... ) ] 
+        INTO num_buckets BUCKETS ]
+    [ LOCATION path ]
+    [ COMMENT table_comment ]
+    [ TBLPROPERTIES ( key1=val1, key2=val2, ... ) ]
+    [ AS select_statement ]
+```
```
Comment (jerryshao) [inline docs/spark-connector/spark-catalog-hive.md:64]: Do you need to add more SQL examples about DDL, DML and DQL to show users how to use it.

File: docs/spark-connector/spark-catalog-hive.md (hunk 4 above)
Comment (FANNG1) [reply docs/spark-connector/spark-catalog-hive.md:9]: no, the configurations are retrivied from Gravitino server.

File: docs/spark-connector/spark-catalog-hive.md (hunk 5 above)
Comment (FANNG1) [reply docs/spark-connector/spark-catalog-hive.md:64]: ok

File: docs/spark-connector/spark-connector.md (hunk 1 above)
Comment (FANNG1) [reply docs/spark-connector/spark-connector.md:48]: done

File: docs/spark-connector/spark-connector.md (hunk 2 above)
Comment (FANNG1) [reply docs/spark-connector/spark-connector.md:54]: done

File: docs/spark-connector/spark-connector.md (hunk 3 above)
Comment (FANNG1) [reply docs/spark-connector/spark-connector.md]: done

File: docs/spark-connector/spark-connector.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+```
+
+3. Execute the spark sql query. 
+
+Suppose there are two catalogs in the metalake `test`, `hive` and `iceberg`, and the table `hive_table1` in the catalog `hive`, and the table `iceberg_table1` in the catalog `iceberg`.
+
+```sql
+select * from hive.db.hive_table1 union all select * from iceberg.db.iceberg_table1;
+use hive;
+select * from db.hive_table1;
```
[the remark this answers is not in the document]
Comment (FANNG1) [reply docs/spark-connector/spark-connector.md]: done

**Review by jerryshao — APPROVED** [review]

**Pull request merged**